nextflow-parsing/NextflowParsingFunction/src/main/java/io/dockstore/nextflowparsing/NextflowUtilities.java

```diff
@@ -0,0 +1,225 @@
+      throw new NextflowParsingException("Could not create Nextflow location", e);
+    }
+    String nextflowTarget = libraryLocation + nextflowFilename;
+    File nextflowTargetFile = new File(nextflowTarget);
+    if (!nextflowTargetFile.exists()) {
+      try {
+        FileUtils.copyURLToFile(nextflowUrl, nextflowTargetFile);
+      } catch (IOException e) {
+        throw new NextflowParsingException("Could not download Nextflow location", e);
+      }
+    }
+    return nextflowTargetFile;
+  }
+
+  /**
+   * Get authors of the workflow.
+   *
+   * @param configuration The Nextflow configuration
+   * @return
+   */
+  public static List<String> getAuthors(Configuration configuration) {
+    List<String> authorsList = new ArrayList<>();
+    if (configuration.containsKey("manifest.author")) {
+      String[] authors =
+          Arrays.stream(configuration.getString("manifest.author").split(","))
+              .map(String::trim)
+              .toArray(String[]::new);
+      authorsList = Arrays.asList(authors);
+    }
+    return authorsList;
+  }
+
+  /**
+   * Get the description of the workflow.
+   *
+   * @param configuration The Nextflow configuration
+   * @return
+   */
+  public static String getDescription(Configuration configuration) {
+    if (configuration.containsKey("manifest.description")) {
+      return configuration.getString("manifest.description");
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Use nextflow to read nextflow configs Nextflow binary assumes workflow is in the working
+   * directory (with no other workflows) TODO: nextflow normally uses DRIP, investigate to see if
+   * this helps us https://github.com/ninjudd/drip
+   *
+   * @param content a file object with the content of nextflow
+   * @return a commons configuration file with the keys from the nextflow config file
+   */
+  public static Configuration grabConfig(File content) {
+    try {
+      final List<String> strings =
+          Arrays.asList(
+              "java", "-jar", getNextflowTargetFile().getAbsolutePath(), "config", "-properties");
+      String stdout = executeNextflowConfig(content.getParentFile(), strings);
+      Properties properties = new Properties();
+      properties.load(new StringReader(stdout));
+      return ConfigurationConverter.getConfiguration(properties);
+    } catch (RuntimeException | IOException | InterruptedException e) {
+      LOG.error("Problem running Nextflow: ", e);
+      throw new NextflowParsingException("Could not run Nextflow", e);
+    }
+  }
+
+  /**
+   * Converts the contents of a config file to a commons config file.
+   *
+   * @param content the content of the config file
+   * @return a commons configuration file with the keys from the nextflow config file
+   */
+  public static Configuration grabConfig(String content) {
+    Path nextflowDir = null;
+    try {
+      // FIXME: this sucks, but we need to ignore includeConfig lines.
```

Comment:
We can probably delete this section once the lambda seems to be working since the clone should get everything